### DISCUSSION TITLE
Is there something like `Entgo` for Rust (and Diesel)?

### DISCUSSION THREAD

**Discussion by frederikhors** [diesel-rs/diesel#3610] [Q&A]
Hi guys!

I'm coming from Go where I used [entgo](https://entgo.io/) to generate automagically structs for SQL queries using GraphQL too.

I'm wondering if there is something like that in Rust (maybe for Diesel).

Basically, given a series of structs with "one to many" or "many to many" relationships, I would like to automatically generate code that allows me to query those relationships in SQL using `AND`, `OR`, `NOT` filters for example.

Here is better explained: https://entgo.io/docs/tutorial-todo-gql-filter-input.

Can you help me understand more?

**Comment by weiznich** [comment]
At some point there was [wundergraph](https://lib.rs/crates/wundergraph) that did something like that (providing a Graphql interface based on a database using rust + diesel) but that's not maintained anymore. Other than that I'm not aware of any automatic solution.
